templates/deployment.yaml

```diff
@@ -42,6 +42,19 @@ spec:
           envFrom:
             - configMapRef:
                 name: {{ include "dsb-client-gateway.fullname" . }}-configmap
+          {{- if .Values.clientgateway.config.basic_auth_enabled }}
+          env:
```

Comment:
you can also load all the data from secret using:
```
envFrom:
            - secretRef:
```
            
no need to load variables one by one